internal/controller/sriov_controller.go

```diff
@@ -0,0 +1,287 @@
+/*
+ Copyright 2025, NVIDIA CORPORATION & AFFILIATES
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Mellanox/spectrum-x-operator/pkg/config"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+)
+
+const (
+	spectrumXSRIOVControllerName = "spectrumxsriovcontroller"
+	ovsNetworkType               = "OVSNetwork"
+	ovsNetworkNamespace          = "default"
+	ovsNetworkInterfaceType      = "dpdk"
+	sriovNodePolicyType          = "SRIOVNetworkNodePolicy"
+	spectrumXSRIOVFinalizer      = "spectrumx.nvidia.com/sriovnetwork"
+)
+
+// SRIOVReconciler reconciles Spectrum-X-Config from ConfigMap
+type SRIOVReconciler struct {
+	client.Client
+	Scheme             *runtime.Scheme
+	ConfigMapNamespace string
+	ConfigMapName      string
+	SriovObjNamespace  string
+}
+
+//+kubebuilder:rbac:groups="",resources=configmaps/finalizers,verbs=update
+//+kubebuilder:rbac:groups=sriovnetwork.openshift.io,resources=ovsnetworks,verbs=get;list;patch;delete;deletecollection
+//+kubebuilder:rbac:groups=sriovnetwork.openshift.io,resources=sriovnetworknodepolicies,verbs=get;list;patch;delete;deletecollection
+
+func (r *SRIOVReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
you can reconcile the config map directly, similar to https://github.com/Mellanox/spectrum-x-operator/blob/main/internal/controller/hostconfig_controller.go
Then you will not need to get the object.

Reply:
In this kind of flow, how can we catch deletion?
We need to clean up objects and remove finalizers

Reply:
same thing, you are checking if deleted at is not zero

Reply:
not a must for this PR